aioxmpp/avatar/service.py

```diff
@@ -575,11 +759,17 @@ def disable_avatar(self):
 
         This is done by setting the avatar metadata node empty.
         """
-        yield from self._check_for_pep()
 
         with (yield from self._publish_lock):
-            yield from self._pubsub.publish(
-                None,
-                namespaces.xep0084_metadata,
-                avatar_xso.Metadata()
-            )
+            try:
+                yield from self._pep.publish(
+                    namespaces.xep0084_metadata,
+                    avatar_xso.Metadata()
+                )
+            finally:
+                if self._synchronize_vcard:
```

Comment:
Here, on the other hand, it makes sense to also try to push the vcard if the other thing failed.

I would suggest to use asyncio.gather or something like that, though, because that will (a) parallelize the request and (b) also ensure that both requests are issued, no matter the exceptions.

Reply:
Yes, I think using gather is what we want.

Reply:
Okay, gather could work with `return_exceptions = True` which is terrible in-band signalling in my opinion ... otherwise gather raises as soon as the first exception happens (so we would get into trouble if only one of the protocols were supported.

Reply:
We don’t need the return values in this case, so it shouldn’t matter.

(If you care about this strongly, you could fake things with:
```python
done, pending = asyncio.wait([coro1, coro2], return_when=asyncio.ALL_COMPLETED)
for fut in done:
    fut.exception() or fut.result()  # needed to avoid warnings
```
… which funnily enough does exactly the same thing like asyncio.gather, probably)